File: languages/go/entity/integrity/proof_anchor.go
package integrity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// ProofAnchor represents a proof anchor.
type ProofAnchor struct {
	AnchorID int64
	Networks []AnchorNetwork
	Root     string
	Status   string
}

func NewProofAnchorFromProto(p *proto.ProofAnchor) ProofAnchor {
	if p == nil {
		return ProofAnchor{}
	}
	return ProofAnchor{
		AnchorID: p.AnchorId,
		Networks: MapAnchorNetworksFromProto(p.Networks),
		Root:     p.Root,
		Status:   p.Status,
	}
}

func (p ProofAnchor) ToProto() *proto.ProofAnchor {
	networks := make([]*proto.AnchorNetwork, len(p.Networks))
	for i, network := range p.Networks {
		networks[i] = network.ToProto()
	}

	return &proto.ProofAnchor{
		AnchorId: p.AnchorID,
		Networks: networks,
		Root:     p.Root,
		Status:   p.Status,
	}
}

File: languages/js/src/entity/integrity/proof_anchor.ts
import * as proto from "../../bridge/proto/integrity_entities";
import { AnchorNetwork } from "./anchor_network";

/**
 * Represents a proof anchor.
 */
export class ProofAnchor {
  anchorID: number;
  networks: AnchorNetwork[];
  root: string;
  status: string;

  /**
   * Constructs a ProofAnchor object with the specified parameters.
   * @param anchorID 
   * @param networks 
   * @param root 
   * @param status 
   */
  constructor(
    anchorID: number,
    networks: AnchorNetwork[],
    root: string,
    status: string
  ) {
    this.anchorID = anchorID;
    this.networks = networks;
    this.root = root;
    this.status = status;
  }

  static fromProto(p: proto.ProofAnchor): ProofAnchor {
    return new ProofAnchor(
      p.anchorId,
      p.networks.map(x => AnchorNetwork.fromProto(x)),
      p.root,
      p.status
    );
  }

  toProto(): proto.ProofAnchor {
    return proto.ProofAnchor.fromPartial({
      anchorId: this.anchorID,
      networks: this.networks.map(n => n.toProto()),
      root: this.root,
      status: this.status
    });
  }
}
